validators/cgroup_validator_linux.go

```diff
@@ -44,10 +47,31 @@ func (c *CgroupsValidator) Name() string {
 }
 
 const (
-	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	cgroupsConfigPrefix      = "CGROUPS_"
+	defaultUnifiedMountpoint = "/sys/fs/cgroup"
+	mountsFilePath           = "/proc/mounts"
 )
 
+// getUnifiedMountpoint will check /proc/mounts and return the cgroup dir
+func getUnifiedMountpoint() string {
+	f, err := os.Open(mountsFilePath)
+	if err != nil {
+		fmt.Printf("error checking %q: %v\n", mountsFilePath, err)
+		return defaultUnifiedMountpoint
+	}
+
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		// example fields: `cgroup2 /sys/fs/cgroup cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime 0 0`
+		fields := strings.Split(scanner.Text(), " ")
+		if len(fields) >= 3 && (fields[2] == "cgroup2" || fields[2] == "cgroup") {
```

Comment:
Which shoud we select the mount point if there are multiple cgroup mount points?

Reply:
The current logic is to return the first `cgroup` or `cgroup2`.

Do you mean that we can return `cgroup2` if we have? And if not, we return the `cgroup` mount point.

Reply:
Updated.